### PR TITLE
Replace cancelPush with AbortSignal

### LINKED ISSUE #147: Use AbortController/*Signal instead of cancelWatch(optional  long id);
https://developer.mozilla.org/en-US/docs/Web/API/AbortController

That is a much more modern way of doing this, but one that didn't exist when `cancelWatch` was added to the spec

### PR DESCRIPTION
**Pull request by kenchris**
Closes #147

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1561,12 +1560,20 @@ <h2>The <dfn>NFCRecordType</dfn> enum</h2>
...
-      steps with <code>"peer"</code> as parameter.
+      For each pending <a>NFC.push</a>:
+      <ol>
+        <li>
+          Stop the instance's <var>timer</var> if it is active.
+        </li>
+        <li>
+          If the instance has already initiated NFC data transfer,
+          continue.
+        </li>
```
Comment (zolkis) [inline index.html:1571]: If we let the transfer continue, I would not stop its timer. 
If a transfer has started and we let it continue, it should finish regularly and perhaps we could fail the abort, then the client code could try later.

File: index.html [hunk 2]
```diff
@@ -1755,200 +1767,228 @@ <h2>The <dfn>NFCWatchMode</dfn> enum</h2>
...
+              <ol>
+                <li>
+                  Stop the instance's <var>timer</var> if it is active.
+                </li>
+                <li>
+                  If the instance has already initiated NFC data transfer,
+                  reject <var>p</var> with
+                  <code>"<a>NoModificationAllowedError</a>"</code> and abort these
+                  steps.
+                </li>
```
Comment (zolkis) [inline index.html:1809]: Here as well, I would put the stop timer step after this one.

**Review by zolkis — COMMENTED** [review]
Nice, thank you. We should double check timer management during aborts.

File: index.html (hunk 1 above)
Comment (kenchris) [reply index.html:1571]: Continue means move to the next item in the foreach...

Comment (zolkis) [reply index.html:1571]: yes, but a transfer is left ongoing and we have no timer left; perhaps it's fine, we just lose the explicit possibility to automatically abort it (using this algorithm), but we don't have much effect over the underlying NFC transfer any more. So the steps are probably "best effort".

Comment (kenchris) [reply index.html:1571]: yes that is how I see it...

**Review by zolkis — APPROVED** [review]
We need to think over the timer management, but the PR is ok in the sense that it's describing a "best effort" abort.

**Pull request merged**